core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -258,7 +258,7 @@ public SystemSessionProperties(
                 booleanProperty(
                         DETERMINE_PARTITION_COUNT_FOR_WRITE_ENABLED,
                         "Determine the number of partitions based on amount of data read and processed by the query for write queries",
-                        false,
+                        queryManagerConfig.isDeterminePartitionCountForWriteEnabled(),
```

Comment:
I think there should be a test that verifies changes to plan based on usage of this session property.

Reply:
possibly, but in this PR I'm just plugging correct property so that users have session properties consistent with feature config

Reply:
cc: @linzebing

Reply:
@raunaqmorarka : turning it on might cause OOM regressions related to partition writes. We do have it enabled for FTE internally while keeping lower bound of partitions to be 50.